public/app/features/plugins/plugin_loader.ts

```diff
@@ -200,11 +217,26 @@ export async function importPluginModule(path: string, version?: string): Promis
       return builtIn;
     }
   }
+
+  // the sandboxing environment code cannot work in nodejs and requires a real browser
+  if (isFrontendSandboxSupported(isAngular)) {
+    return importPluginModuleInSandbox({ pluginId });
```

Comment:
In case the sandboxing doesn't work inside nodejs, what would be the best option to add tests for it? Probably adding some E2E tests is the best we can do? (As this is a quite complex piece of logic and also really important we should have some automated tests for it.)

Reply:
e2e is the only way to know. There's no way this sandbox will work in nodejs due to its nature.